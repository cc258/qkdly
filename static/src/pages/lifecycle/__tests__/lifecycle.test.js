import React from 'react'
import Lifecycle from '../lifecycle'
import { shallow } from 'enzyme';

import axios from 'axios'
import MockAdapter from "axios-mock-adapter"

describe('<Lifecycle />', () => {
	beforeEach(function () {
		const mock = new MockAdapter(axios);
		mock.onGet("https://api.github.com/users").reply(200, {
			users: [{ login: "John Smith" }],
		});

		mock.onGet("/users/12345").reply(200, {
			users: [{ login: "John Smith" }],
		});
	})

	it('render scorret', () => {
		const wrapper = shallow(<Lifecycle />)
		const ins = wrapper.instance()
		expect(wrapper).toMatchSnapshot();

		ins.getApi();

	})

});