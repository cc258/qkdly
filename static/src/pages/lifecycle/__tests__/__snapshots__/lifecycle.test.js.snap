// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Lifecycle /> render scorret 1`] = `
<div
  style={
    Object {
      "border": "5px solid red",
      "padding": "5px",
    }
  }
>
  <p>
    计数器
    :
    0
  </p>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <ul />
  <SubCounter
    number={0}
  />
  <button
    onClick={[Function]}
  />
</div>
`;
